services/app-api/src/common-code/contractAndRates/ContractProvisions.ts

```diff
@@ -0,0 +1,206 @@
+import {
+    ModifiedProvisionsAmendmentRecord,
+    ModifiedProvisionsBaseContractRecord,
+    ModifiedProvisionsCHIPRecord,
+} from '../../constants/modifiedProvisions'
+import {
+    type CHIPProvisionType,
+    type MedicaidBaseProvisionType,
+    type MedicaidAmendmentProvisionType,
+    type GeneralizedProvisionType,
+    provisionCHIPKeys,
+    modifiedProvisionMedicaidBaseKeys,
+    modifiedProvisionMedicaidAmendmentKeys,
+    isCHIPProvision,
+    isMedicaidAmendmentProvision,
+    isMedicaidBaseProvision,
+} from './ModifiedProvisions'
+import {
+    isBaseContract,
+    isCHIPOnly,
+    isContractAmendment,
+    isContractWithProvisions,
+    getLastContractSubmission,
+} from './contractData'
+import type { ContractType, UnlockedContractType } from '../../domain-models'
+
+/*
+    Each provision key represents a Yes/No question asked on Contract Details. 
+    This is a set of helper functions that each take in a submission and return provisions related data.
+
+    There are currently three distrinct variants of the provisions: 
+    1. For CHIP amendment
+    2. For non CHIP base contract
+    3. For non CHIP contract amendment
+
+    See also ModifiedProvisions.ts
+*/
+
+// Returns the list of provision keys that apply for given submission variant
+const generateApplicableProvisionsList = (
```

Comment:
nit: These functions look good. I wonder if it would be helpful to add unit tests for these functions. I will be hard to manually test.

Reply:
Most of these are already tested in `provisions.test.tsx` already in app-web. This is copypasta because we don't yet have common code fixed.

Reply:
Ah, I see why this probably wasn't tested. We lack a lot of the mock functions on the backend that the frontend has for contract and rates. I started make some but pausing because it seems like a pretty big task. Let me see if I can knock out smaller scope issues first.